lib/galaxy/datatypes/binary.py

```diff
@@ -1017,30 +1017,30 @@ def set_meta(self, dataset, overwrite=True, **kwd):
             f = h5py.File(dataset.file_name)
             attributes = dict(f.attrs.items())
 
-            dataset.metadata.id = attributes['id']
-            dataset.metadata.format_url = attributes['format-url']
+            dataset.metadata.id = util.unicodify(attributes['id'])
+            dataset.metadata.format_url = util.unicodify(attributes['format-url'])
             if 'format-version' in attributes:  # biom 2.1
-                dataset.metadata.format_version = '.'.join(map(str, list(attributes['format-version'])))
+                dataset.metadata.format_version = '.'.join((util.unicodify(_) for _ in list(attributes['format-version'])))
```

Comment:
These are numbers in a numpy.ndarray, so `str()` should be enough.
```suggestion
                dataset.metadata.format_version = '.'.join(str(_) for _ in attributes['format-version'])
```